src/EnergyPlus/HeatBalanceAirManager.cc

```diff
@@ -4902,6 +4903,36 @@ void ReportZoneMeanAirTemp(EnergyPlusData &state)
     // PURPOSE OF THIS SUBROUTINE:
     // This subroutine updates the report variables for the AirHeatBalance.
 
+    if (state.dataHeatBalAirMgr->CalcExtraReportVarMyOneTimeFlag) {
+        for (auto const *reqVar : state.dataOutputProcessor->reqVars) {
+            if (reqVar->name == "ZONE WETBULB GLOBE TEMPERATURE") {
+                if (reqVar->key.empty()) {
+                    for (int ZoneLoop = 1; ZoneLoop <= state.dataGlobal->NumOfZones; ++ZoneLoop) {
+                        if (state.dataHeatBal->doSpaceHeatBalanceSimulation) {
+                            for (int spaceNum : state.dataHeatBal->Zone(ZoneLoop).spaceIndexes) {
+                                auto &thisSpaceAirRpt = state.dataHeatBal->spaceAirRpt(spaceNum);
+                                thisSpaceAirRpt.ReportWBGT = true;
+                            }
+                        } else {
+                            auto &thisZnAirRpt = state.dataHeatBal->ZnAirRpt(ZoneLoop);
+                            thisZnAirRpt.ReportWBGT = true;
+                        }
+                    }
+                } else {
+                    if (state.dataHeatBal->doSpaceHeatBalanceSimulation) {
+                        int spaceNum = Util::FindItemInList(reqVar->name, state.dataHeatBal->space);
```

Comment:
These need to search on `reqVar->key`, not `name`. If you run a test file with a zone name in the output variable object, it won't work here. And this needs to test for `spaceNum > 0` to prevent a subscript error. Same comments apply to zone below.

Reply:
I'll search the "key" field here for the space and zone name instead of the "name" field, and add a check for the space and zone index > 0.